src/wp-admin/includes/export.php

```diff
@@ -144,6 +144,40 @@ function export_wp( $args = array() ) {
 	// Grab a snapshot of post IDs, just in case it changes during the export.
 	$post_ids = $wpdb->get_col( "SELECT ID FROM {$wpdb->posts} $join WHERE $where" );
 
+	// Get IDs for the attachments of each post, unless all content is already being exported.
+	if ( ! in_array( $args['content'], array( 'all', 'attachment' ), true ) ) {
+		foreach ( array_chunk( $post_ids, 20 ) as $chunk ) {
```

Comment:
There's already a [chunking approach in the same file](https://github.com/WordPress/wordpress-develop/blob/6.4/src/wp-admin/includes/export.php#L544):

```
// Fetch 20 posts at a time rather than loading the entire table into memory.
while ( $next_posts = array_splice( $post_ids, 0, 20 ) ) {
```

Could that approach also be used here for consistency? 

Are there benefits to this approach over the other?

Reply:
The approach you mentioned actually is similar, but it modifies the array instead of creating new ones. I went ahead and used it for consistency but I also needed to create a copy of the `$post_ids` so it wouldn't get modified.